README.adoc

```diff
@@ -80,25 +80,52 @@ checkout([$class: 'GitSCM',
 
 ==== Checkout with ssh and a private key credential
 
-Insert example here
+Checkout from the git client plugin source repository using https protocol, ssh private credentials, and a non-default branch. 
+The git plugin supports private key credentials provided by the link:https://plugins.jenkins.io/credentials[Jenkins credentials plugin].
+
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: 'master']],
+          userRemoteConfigs: [[credentialsId: 'e1476d45-b77d-4af6-a80f-e2fe95eca822', url: 'https://github.com/jenkinsci/git-plugin.git']]])
+---
 
 ==== Checkout with https and a username/password credential
 
-Insert example here
+Checkout from the git client plugin source repository using https protocol, username/password credentials, and a non-default branch. The git plugin supports username/password credentials provided by the link:https://plugins.jenkins.io/credentials[Jenkins credentials plugin].
+
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: 'jenkinsfile']],
+          userRemoteConfigs: [[credentialsId: 'e1476d45-b77d-4af6-a80f-e2fe95eca822', url: 'https://github.com/jenkinsci/git-client-plugin.git']]])
+---
 
 ==== Checkout with git large file support enabled
 
-Enable git lfs pull after checkout, create the snippet
+Checkout from the git client plugin source repository using https no credentials, a non-default branch, and git large file support extension link:https://git-lfs.github.com 
+[Enable git large file support] for the workspace by pulling large files after the checkout completes.
 
-==== Add a repository browser for Bitbucket
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: 'disposer']],
+          extensions: [[$class: 'GitLFSPull']],
+          userRemoteConfigs: [[url: 'https://github.com/jenkinsci/git-client-plugin.git']]])
+---
 
-==== Checkout without fetching tags (advanced clone behavior)
 
-==== Checkout with a shallow clone to reduce data traffic
+==== Checkout without fetching tags (advanced clone behavior)
 
-==== Checkout with a narrow refspec (needs more reading)
+Checkout from the git client plugin source repository using https no credentials, a default branch, and without tags saving time and disk space when you want to access only what is specified by the refspec, without considering any repository tags.
```

Comment:
```suggestion
Checkout from the git client plugin source repository using https with no credentials, a default branch, and without tags.
This can save time and disk space when you want to access the repository without considering tags.
```